utility/src/test/java/org/oppia/util/datetime/DateTimeUtilTest.kt

```diff
@@ -0,0 +1,157 @@
+package org.oppia.util.datetime
+
+import android.app.Application
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.ObsoleteCoroutinesApi
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.util.logging.EnableConsoleLog
+import org.oppia.util.logging.EnableFileLog
+import org.oppia.util.logging.GlobalLogLevel
+import org.oppia.util.logging.LogLevel
+import org.oppia.util.system.OppiaClock
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import org.robolectric.annotation.Config
+import javax.inject.Inject
+import javax.inject.Qualifier
+import javax.inject.Singleton
+
+private const val EVENING_TIMESTAMP = 1556061720000
+private const val MORNING_TIMESTAMP = 1556094120000
+private const val AFTERNOON_TIMESTAMP = 1556029320000
+
+/** Tests for [DateTimeUtil]. */
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class DateTimeUtilTest {
+
+  @Inject lateinit var dateTimeUtil: DateTimeUtil
+  @Inject lateinit var context: Context
+  @Inject lateinit var oppiaClock: OppiaClock
+
+  // https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/
+  @ObsoleteCoroutinesApi
+  private val testThread = newSingleThreadContext("TestMain")
```

Comment:
change code to ` @ObsoleteCoroutinesApi private val testThread = newSingleThreadContext("TestMain")`
 also Nit: ``Reference document: https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/``

Reply:
Updated.